contracts/lido_cosmos_hub/src/contract.rs

```
@@ -155,18 +161,43 @@ pub fn execute_remove_guardians(
         .add_attributes(guardians.iter().map(|g| attr("value", g))))
 }
 
-pub fn execute_pause_contracts(deps: DepsMut, _env: Env, info: MessageInfo) -> StdResult<Response> {
+pub fn execute_pause_contracts(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    duration: u64,
+) -> StdResult<Response> {
     let config = CONFIG.load(deps.storage)?;
     if !(info.sender == config.creator || GUARDIANS.has(deps.storage, info.sender.to_string())) {
         return Err(StdError::generic_err("unauthorized"));
     }
 
+    if duration == 0 {
+        return Err(StdError::generic_err(
+            "pause duration should be greater than zero",
+        ));
+    }
+
+    let duration = duration.min(MAX_PAUSE_DURATION);
```

Looks like unexpected behaviour for me. What if one requests a pause for two weeks and contract silently only gets paused for a week? I mean, it doesn't sounds that bad, but maybe we should just return an error which says "hey dude, you can only pause as long as up to 1 week!".

That's mostly a question, not a recommendation.

---

it's not done silently, but we return pause duration and end time as attributes in the response

---

I agree. It might be a better idea to explicitly return an error here.

---

Well, that's still far from perfect from my point of view. Let's wait until team says their opinion.